src/app/gps/qgsgpsinformationwidget.cpp

```diff
@@ -1461,6 +1487,38 @@ void QgsGpsInformationWidget::timestampFormatChanged( int )
   mLblTimeZone->setEnabled( enabled );
 }
 
+void QgsGpsInformationWidget::cursorCoordinateChanged( const QgsPointXY &point )
+{
+  try
+  {
+    mLastCursorPosWgs84 = mCanvasToWgs84Transform.transform( point );
+    updateGpsDistanceStatusMessage();
+  }
+  catch ( QgsCsException & )
+  {
+
+  }
+}
+
+void QgsGpsInformationWidget::updateGpsDistanceStatusMessage()
+{
+  if ( !mNmea )
+    return;
+
+  QgsDistanceArea da;
```

Comment:
Out of curiosity, is that QgsDistanceArea expensive to create (setEllipsoid, setSourceCrs, etc.)? If the answer is yes, should it instead be a member of the QgsGpsInformationWidget class, and update the ellipsoid / source CRS through project signals?

Reply:
Good point, I'll do that

Reply:
Done